### PR TITLE
Enable lua support for fluent-bit

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Updated version of https://github.com/microsoft/azurelinux/pull/10690 with rebase against latest changes.

###### Change Log  

- Update fluent-bit to 3.1.9
- Enable lua support for fluent-bit
- Remove vendored CVE patches that have been applied upstream
  - https://github.com/fluent/fluent-bit/commit/f0416917e47b06a537abc2e9f40390a9e3918fb2 added in v3.1.0
  - https://github.com/fluent/fluent-bit/commit/7e2f3f088aade6a30cc16130bcd707b79d11b977 added in v3.1.7

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- https://microsoft.visualstudio.com/DefaultCollection/OS/_queries/edit/54713209

###### Test Methodology

- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=673523&view=results
- Installed in azl3 container and ran some basic smoke tests from https://docs.fluentbit.io/manual/stream-processing/getting-started/hands-on
- Installed in azl3 marketplace vm

### CODE REVIEW

**Review by paulmey — APPROVED** [review]
LGTM

File: SPECS/fluent-bit/fluent-bit.spec [hunk 1]
```diff
@@ -26,6 +24,7 @@ BuildRequires:  pkgconfig
 BuildRequires:  systemd-devel
 BuildRequires:  systemd-rpm-macros
 BuildRequires:  zlib-devel
+BuildRequires:  luajit-devel
```
Comment (PawelWMS) [inline SPECS/fluent-bit/fluent-bit.spec:27]: Since the list was already in alphabetical order, could you move the new line up a bit?

**Review by PawelWMS — APPROVED** [review]

**Comment by dmcilvaney** [comment]
Tested by running (some) of the bits here https://docs.fluentbit.io/manual/stream-processing/getting-started/hands-on

**Comment by dmcilvaney** [comment]
Unclear why the buddybuild install test fails, I am able to install successfully.

**Review by jslobodzian — APPROVED** [review]

**Review by paulmey — APPROVED** [review]

**Pull request merged**